File: frontend/web/services/services/services.ts
import { axiosInstance } from "@/services";
import { SERVICE_ERROR_UNKNOWN } from "@/config/services";
import { ServiceReturn } from "@/types/api";
import { Service } from "@/types/services";

const getAll = async (
	has?: "actions" | "reactions",
	isAuthenticator: boolean = false,
): Promise<ServiceReturn<Service[]>> => {
	try {
		// eslint-disable-next-line no-nested-ternary
		const query = has !== undefined ? `?has=${has}` : isAuthenticator ? "?isAuthenticator=true" : "";
		const response = await axiosInstance.get<Service[]>(`/services${query}`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

const getOne = async (serviceId: string): Promise<ServiceReturn<Service>> => {
	try {
		const response = await axiosInstance.get<Service>(`/services/${serviceId}`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

export { getAll, getOne };

File: frontend/mobile/lib/services/services/services.dart
import 'package:area_mobile/services/dio.dart';
import 'package:area_mobile/types/services.dart';

Future<ServiceReturn<List<Service>>> getAll([String? has]) async {
  try {
    final query = has != null ? "?has=$has" : "";
    final response = await dio.get<List<dynamic>>('/services$query');
    return ServiceReturn(
        data: response.data!.map((e) => Service.fromJson(e)).toList());
  } catch (e) {
    return ServiceReturn(error: e.toString());
  }
}

Future<ServiceReturn<Service>> getOne(String serviceId) async {
  try {
    final response = await dio.get<dynamic>('/services/$serviceId');
    return ServiceReturn(data: Service.fromJson(response.data));
  } catch (e) {
    return ServiceReturn(error: e.toString());
  }
}
